### DISCUSSION TITLE
How to enable cookieless before consent and then disable it after consent?

### DISCUSSION THREAD

**Discussion by michacassola** [AmauriC/tarteaucitron.js#1025] [Q&A] [answered]
Hi, thank you very much for this project!

I would like to implement matomo tracking of users without cookies as long as they have not given consent yet and also after they have given consent as described here: https://matomo.org/faq/new-to-piwik/how-can-i-still-track-a-visitor-without-cookies-even-if-they-decline-the-cookie-consent/
How would I achieve something like that with Tarte au Citron?

**Comment by AmauriC** [comment, marked as answer]
tarteaucitron can manage this, I'll do it ;)
https://github.com/AmauriC/tarteaucitron.js/issues/1028

**Comment by michacassola** [comment]
Is it possible to also do this with Matomo Tracking and PostHog Tracking?
Both should start cookieless and switch to cookie based after consent.
Possibly using also Matomo Tag Manager.
My understanding is limited, but I think it is not easy to delete cookies being set by Matomo Tag Manager, so this kind of thing would need to be done for everything by intself and using a Tag Manager is not adviced?
